skrf/network.py

```diff
@@ -3223,33 +3223,31 @@ def step_response(self, window='hamming', n=None, pad=1000):
         y : class:`numpy.ndarray`
             Step response
 
+        Raises
+        ------
+        ValueError
+            If used with an Network with more than one port
+
+        NotImplementedError
+            If used with non equidistant sampled frequency vector
+
         See Also
         -----------
             impulse_response
             extrapolate_to_dc
         """
-        if self.nports != 1:
-            raise ValueError('Only one-ports are supported')
+        if self.frequency.sweep_type != 'lin':
+            raise NotImplementedError(
+                'Unable to transform non equidistant sampled points to time domain')
+        
         if self.frequency.f[0] != 0:
             warnings.warn(
                 "Frequency doesn't begin from 0. Step response will not be correct.",
                 RuntimeWarning
             )
-        if n is None:
-            # Use zero-padding specification. Note that this does not allow n odd.
-            n = 2 * (self.frequency.npoints + pad - 1)
 
-        fstep = self.frequency.step
-        if n % 2 == 0:
-            t = npy.flipud(npy.linspace(.5 / fstep, -.5 / fstep, n, endpoint=False))
-        else:
-            t = npy.flipud(npy.linspace(.5 / fstep, -.5 / fstep, n + 1, endpoint=False))
-            t = t[:-1]
-        if window != None:
-            w = self.windowed(window=window, normalize=False, center_to_dc=True)
-        else:
-            w = self
-        return t, npy.cumsum(mf.irfft(w.s, n=n).flatten())
+        t, y = self.impulse_response(window=window, n=n, pad=pad, bandpass=False)
+        return t, npy.cumsum(y)
```

Comment:
Did you check this is correct is any case (I didn't)?

Reply:
Only for an linear sweep starting at DC, otherwise an Exception is raised anyway.
Check this [gist](https://gist.github.com/FranzForstmayr/0f22a87c39aa80e7fb8b0ff3256b7a4a#file-step-py):